instrumentation/host/example/main.go

```diff
@@ -42,14 +43,18 @@ func initMeter() *controller.Controller {
 			exporter,
 		),
 		controller.WithExporter(exporter),
+		controller.WithCollectPeriod(time.Second*3),
 	)
 	pusher.Start(context.Background()) //nolint:errcheck
 	global.SetMeterProvider(pusher.MeterProvider())
 	return pusher
 }
 
 func main() {
-	defer handleErr(initMeter().Stop(context.Background()))
+	pusher := initMeter()
+	defer func(pusher *controller.Controller, ctx context.Context) {
+		handleErr(pusher.Stop(ctx))
+	}(pusher, context.Background())
```

Comment:
I find this example code a little too hard to read.
What about something like:

```go
func main() {
	exporter, err := stdout.New(stdout.WithPrettyPrint())
	if err != nil {
		log.Fatalln("failed to initialize metric stdout exporter:", err)
	}
	pusher := controller.New(
		processor.New(
			simple.NewWithInexpensiveDistribution(),
			exporter,
		),
		controller.WithExporter(exporter),
		controller.WithCollectPeriod(3*time.Second),
	)
	if err := pusher.Start(context.Background()); err != nil {
		log.Fatalln("failed to start the metric controller:", err)
	}
	global.SetMeterProvider(pusher.MeterProvider())

	if err := host.Start(); err != nil {
		log.Fatalln("failed to start host instrumentation:", err)
	}

	ctx, cancel := newOSSignalContext()
	defer cancel()
	<-ctx.Done()

	if err := pusher.Stop(context.Background()); err != nil {
		log.Fatalln("failed to stop the metric controller:", err)
	}
}

func newOSSignalContext() (context.Context, func()) {
	// trap Ctrl+C and call cancel on the context
	ctx, cancel := context.WithCancel(context.Background())
	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	go func() {
		select {
		case <-c:
			cancel()
		case <-ctx.Done():
		}
	}()

	return ctx, func() {
		signal.Stop(c)
		cancel()
	}
}
```

Reply:
^^ This looks good.